packages/perseus/src/widgets/categorizer/categorizer.tsx

```diff
@@ -92,8 +92,11 @@ export class Categorizer
         const isMobile = this.props.apiOptions.isMobile;
         let indexedItems = this.props.items.map((item, n) => [item, n]);
         if (this.props.randomizeItems) {
-            // @ts-expect-error - TS4104 - The type 'readonly (string | number)[][]' is 'readonly' and cannot be assigned to the mutable type '(string | number)[][]'. | TS2345 - Argument of type 'number | null | undefined' is not assignable to parameter of type 'number | RNG'.
-            indexedItems = Util.shuffle(indexedItems, this.props.problemNum);
+            // @ts-expect-error - TS4104 - The type 'readonly (string | number)[][]' is 'readonly' and cannot be assigned to the mutable type '(string | number)[][]'.
+            indexedItems = CoreUtil.shuffle(
+                indexedItems,
+                this.props.problemNum as any,
+            );
```

Comment:
It feels like there could be an opportunity to tighten up our types here. Like can `problemNum` really be `null | undefined`? Would that break `shuffle`?

But if we don't want to go down that rabbit hole, maybe:

``` TS
        let indexedItems: ReadonlyArray<Readonly<[string, number]>> =
            this.props.items.map((item, n) => [item, n]);
        if (this.props.randomizeItems) {
            indexedItems = Util.shuffle(
                indexedItems,
                this.props.problemNum as number,
            );
        }
```

Reply:
Could we avoid the `as any` by doing this?

```ts
indexedItems = CoreUtil.shuffle(
    indexedItems,
    Number(this.props.problemNum),
)
```

or this?

```ts
indexedItems = CoreUtil.shuffle(
    indexedItems,
    this.props.problemNum ?? 0,
)
```

Either of those will handle `null` and `undefined` values, converting them to 0. The problem with using `as any` is that it's unsafe; `CoreUtils.shuffle` says it needs a number as the second argument, not null or undefined. The `as any` is only needed here because we are telling `shuffle` to do something that it explicitly says it does not support.

Alternatively, we could change the types of `CoreUtils.shuffle` to accept `null | undefined` for the second argument.

Reply:
I worry about changing what is passed into the function to zero, mostly because I don't know how it handled receiving undefined/null answers before. I also worry about changing it from `as any` to as `as number` but that seems like it's more type focused, so I went with that suggestion :)

Did a little bit of testing, and if it receives undefined it does shuffle it, but the shuffle doesn't change. If it receives null, it does a different shuffle, but it doesn't change either. Passing in `0` gave the same result as `null`.